### PR TITLE
yolov8n Instance segmentation tutorial

### PR DESCRIPTION
**Pull request by samuel-wj-chapman**
## Pull Request Description:
Notebook utilities required for instance segmentation quantisation tutorial.

## Checklist before requesting a review:
- [x] I set the appropriate labels on the pull request.
- [x] I have added/updated the release note draft (if necessary).
- [x] I have updated the documentation to reflect my changes (if necessary).
- [x] All function and files are well documented. 
- [x] All function and classes have type hints.
- [x] There is a licenses in all file.
- [x] The function and variable names are informative. 
- [x] I have checked for code duplications.
- [x] I have added new unittest (if necessary).

### CODE REVIEW

File: tutorials/notebooks/imx500_notebooks/pytorch/pytorch_yolov8n_seg_for_imx500.ipynb [hunk 1]
```diff
@@ -0,0 +1,619 @@
...
+        "id": "7c7fa04c9903736f",
+        "outputId": "51eab6ab-4821-4cd4-9210-3561fd15a09c"
+      },
+      "outputs": [],
+      "source": [
+        "!pip install -q torch\n",
+        "!pip install onnx\n",
+        "!pip install -q pycocotools\n",
+        "!pip install 'huggingface-hub>=0.21.0'\n",
+        "!pip install --pre 'sony-custom-layers-dev>=0.2.0.dev5'"
```
Comment (Idan-BenAmi) [inline tutorials/notebooks/imx500_notebooks/pytorch/pytorch_yolov8n_seg_for_imx500.ipynb:60]: why do you need sony-custom-layers? I think you can remove this line.

File: tutorials/notebooks/imx500_notebooks/pytorch/pytorch_yolov8n_seg_for_imx500.ipynb [hunk 2]
```diff
@@ -0,0 +1,619 @@
...
+      "cell_type": "code",
+      "execution_count": null,
+      "id": "VLwCbC2_szpy",
+      "metadata": {
+        "id": "VLwCbC2_szpy"
+      },
+      "outputs": [],
+      "source": [
+        "from tutorials.mct_model_garden.evaluation_metrics.coco_evaluation import evaluate_yolov8_segmentation\n",
+        "evaluate_yolov8_segmentation(quant_model, data_dir='coco', data_type='val2017', img_ids_limit=100, output_file='results_g_quant.json')"
```
Comment (Idan-BenAmi) [inline tutorials/notebooks/imx500_notebooks/pytorch/pytorch_yolov8n_seg_for_imx500.ipynb:466]: quant_model --> gptq_quant_model

File: tutorials/mct_model_garden/models_pytorch/yolov8/yolov8.py [hunk 3]
```diff
@@ -519,3 +514,106 @@ def yolov8_pytorch_pp(model_yaml: str,
...
+        bs = p.shape[0]  # batch size
+
+        mc = torch.cat([self.cv4[i](x[i]).view(bs, self.nm, -1) for i in range(self.nl)], 2)  # mask coefficients
+        y_bb, y_cls = self.detect(self, x)
+        
+        return y_bb, y_cls, mc, p   
+
+
+
+class SegmentationModelPyTorch(nn.Module, PyTorchModelHubMixin):
```
[the remark this answers is not in the document]
Comment (samuel-wj-chapman) [reply tutorials/mct_model_garden/models_pytorch/yolov8/yolov8.py]: have merged the two as ModelPytorch

**Comment by samuel-wj-chapman** [comment]
All relevant changes should be added

File: tutorials/mct_model_garden/models_pytorch/yolov8/yolov8.py [hunk 4]
```diff
@@ -14,7 +14,7 @@
 
 The code is organized as follows:
 - Classes definitions of Yolov8n building blocks: Conv, Bottleneck, C2f, SPPF, Upsample, Concaat, DFL and Detect
-- Detection Model definition: DetectionModelPyTorch
+- Detection Model definition: DetectionModelKeras
```
Comment (Idan-BenAmi) [inline tutorials/mct_model_garden/models_pytorch/yolov8/yolov8.py:17]: looks like a typo? Keras --> PyTorch

File: tutorials/notebooks/imx500_notebooks/pytorch/pytorch_yolov8n_for_imx500.ipynb [hunk 5]
```diff
@@ -343,58 +350,51 @@
...
+   "id": "546ff946af81702b",
+   "metadata": {
+    "collapsed": false,
+    "tags": [
+     "long_run"
+    ]
+   },
    "outputs": [],
    "source": [
     "mct.exporter.pytorch_export_model(model=gptq_quant_model_pp,\n",
```
Comment (Idan-BenAmi) [inline tutorials/notebooks/imx500_notebooks/pytorch/pytorch_yolov8n_for_imx500.ipynb:379]: I think we better export the model without the "_pp", otherwise it can be confusing.. we need to emphasize that the "_pp" is for evaluation purpose.

File: tutorials/notebooks/imx500_notebooks/pytorch/pytorch_yolov8n_for_imx500.ipynb [hunk 6]
```diff
@@ -128,9 +128,9 @@
    "metadata": {},
```
Comment (Idan-BenAmi) [inline tutorials/notebooks/imx500_notebooks/pytorch/pytorch_yolov8n_for_imx500.ipynb:128]: need to update the link for google colab.

File: tutorials/mct_model_garden/models_pytorch/yolov8/yolov8.py [hunk 7]
```diff
@@ -519,3 +448,108 @@ def yolov8_pytorch_pp(model_yaml: str,
...
+        p = self.proto(x[0])  # mask protos
+        bs = p.shape[0]  # batch size
+
+        mc = torch.cat([self.cv4[i](x[i]).view(bs, self.nm, -1) for i in range(self.nl)], 2)  # mask coefficients
+        y_bb, y_cls = self.detect(self, x)
+        
+        return y_bb, y_cls, mc, p   
+
+
+class ModelPyTorch(nn.Module, PyTorchModelHubMixin):
```
Comment (Idan-BenAmi) [inline tutorials/mct_model_garden/models_pytorch/yolov8/yolov8.py:497]: I wonder if the renaming of the module will now work with the detection model.. we need to check

**Comment by samuel-wj-chapman** [comment]
broken link due to pre-emptive link.

**Review by Idan-BenAmi — APPROVED** [review]
